File: source_funcs/go_transcoder/COUNT_CHARACTERS_STRING_DISTANCE_ENGLISH_ALPHABETS/COUNT_CHARACTERS_STRING_DISTANCE_ENGLISH_ALPHABETS.go
package main

import "math"

func f_gold(str []byte) int {
	var (
		result int = 0
		n      int = int(len(str))
	)
	for i := int(0); i < n; i++ {
		for j := int(i + 1); j < n; j++ {
			if math.Abs(float64(str[i]-str[j])) == math.Abs(float64(i-j)) {
				result++
			}
		}
	}
	return result
}

File: rust_funcs/go_transcoder/COUNT_CHARACTERS_STRING_DISTANCE_ENGLISH_ALPHABETS/COUNT_CHARACTERS_STRING_DISTANCE_ENGLISH_ALPHABETS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(str: Vec<char>) -> i32 {
    let mut result: i32= 0;
    let n = str.len() as i32;
    for i in 0..n{
        for j in (i + 1)..n{
            if (str[i as usize] as i32 - str[j as usize] as i32).abs() == (i - j).abs(){
                result += 1; 
            }
        }
    }
   result 
}
